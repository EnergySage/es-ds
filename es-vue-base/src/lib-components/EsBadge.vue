<template>
    <b-badge
        :variant="variant"
        v-bind="$attrs"
        v-on="$listeners">
        <!--
            @slot Badge Content
            @binding {string} text or html of the badge content
        -->
        <slot />
    </b-badge>
</template>

<script lang="js">
import { BBadge } from 'bootstrap-vue';

export default {
    name: 'EsBadge',
    components: {
        BBadge,
    },
    props: {
        /**
         * Badge Variant
         *
         */
        variant: {
            type: String,
            default: 'primary',
            required: false,
        },
    },
};
</script>

<style lang="scss">
@import '~@energysage/es-bs-extends/scss/includes';

.badge {
    line-height: 1rem;
}

.badge-primary {
    /* hard-coded background-color due to no satisfactory variation of defined colors */
    background: #CCE5EF;
    color: darken($primary, 15%);
}

.badge-primary:hover {
    background: $primary;
    color: $white;
}

.badge-success {
    /* hard-coded background-color due to no satisfactory variation of defined colors */
    background: #CCE6DA;
    color: darken($success, 14%);
}

.badge-success:hover {
    background: $success;
    color: $white;
}

.badge-danger {
    background: lighten($danger, 42%);
    color: darken($danger, 23%);
}

.badge-danger:hover {
    background: $danger;
    color: $white;
}

.badge-warning {
    background: lighten($warning, 26%);
    color: darken($warning, 41%);
}

.badge-warning:hover {
    background: $warning;
}

.badge-info {
    background: lighten($info, 59%);
    color: $indigo;
}

.badge-info:hover {
    background: $info;
    color: $white;
}

.badge-disabled {
    background: $gray-200;
    color: $gray-700;
}

.badge-disabled:hover {
    background: $gray-400;
}

.badge-dark {
    background: $gray-400;
    color: $gray-800;
}

.badge-dark:hover {
    background: $dark;
    color: $white;
}

</style>
