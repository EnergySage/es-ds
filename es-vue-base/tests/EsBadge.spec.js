import { mount } from '@vue/test-utils';
import EsBadge from '@/src/lib-components/EsBadge.vue';
import jestVue from '@/tests/jest.vue.config';

describe('EsBadge', () => {
    // Basic test; should exist for most components
    const variantOptions = ['primary', 'secondary', 'danger', 'warning', 'info', 'light', 'dark', 'link'];
    const outlineOptions = [true, false];
    const slots = {
        default: 'Main Content',
    };
    const paramaters = [];
    variantOptions.forEach((variant) => {
        outlineOptions.forEach((outline) => {
            paramaters.push({
                variant,
                outline,
            });
        });
    });

    test.each(paramaters)(
        '<es-badge variant="$variant" outline="$outline" />',
        async ({
            variant, outline,
        }) => {
            const wrapper = mount(EsBadge, {
                ...jestVue,
                slots,
                propsData: {
                    variant,
                    outline,
                },
            });
            const a11y = await axe(wrapper.element);

            expect(wrapper.vm).toBeTruthy();
            expect(wrapper.html()).toMatchSnapshot();
            expect(a11y).toHaveNoViolations();
        },
    );

    test('<EsBadge />', async () => {
        const wrapper = mount(EsBadge, {
            ...jestVue,
        });
        const a11y = await axe(wrapper.element);

        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();
        expect(a11y).toHaveNoViolations();
    });

    test('check for <EsBadge />', async () => {
        const wrapper = mount(EsBadge, {
            ...jestVue,
        });

        expect(wrapper.find('.badge').exists()).toBe(true);
    });

    test('Check default text is equivalent to slot default', async () => {
        const wrapper = mount(EsBadge, {
            ...jestVue,
            slots,
        });

        expect(wrapper.text()).toBe('Main Content');
    });
});
