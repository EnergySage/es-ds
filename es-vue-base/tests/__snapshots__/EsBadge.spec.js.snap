// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EsBadge <EsBadge /> 1`] = `"<span class=\\"badge badge-primary\\"></span>"`;

exports[`EsBadge <es-badge variant="danger" outline="false" /> 1`] = `"<span class=\\"badge badge-danger\\">Main Content</span>"`;

exports[`EsBadge <es-badge variant="danger" outline="true" /> 1`] = `"<span outline=\\"true\\" class=\\"badge badge-danger\\">Main Content</span>"`;

exports[`EsBadge <es-badge variant="dark" outline="false" /> 1`] = `"<span class=\\"badge badge-dark\\">Main Content</span>"`;

exports[`EsBadge <es-badge variant="dark" outline="true" /> 1`] = `"<span outline=\\"true\\" class=\\"badge badge-dark\\">Main Content</span>"`;

exports[`EsBadge <es-badge variant="info" outline="false" /> 1`] = `"<span class=\\"badge badge-info\\">Main Content</span>"`;

exports[`EsBadge <es-badge variant="info" outline="true" /> 1`] = `"<span outline=\\"true\\" class=\\"badge badge-info\\">Main Content</span>"`;

exports[`EsBadge <es-badge variant="light" outline="false" /> 1`] = `"<span class=\\"badge badge-light\\">Main Content</span>"`;

exports[`EsBadge <es-badge variant="light" outline="true" /> 1`] = `"<span outline=\\"true\\" class=\\"badge badge-light\\">Main Content</span>"`;

exports[`EsBadge <es-badge variant="link" outline="false" /> 1`] = `"<span class=\\"badge badge-link\\">Main Content</span>"`;

exports[`EsBadge <es-badge variant="link" outline="true" /> 1`] = `"<span outline=\\"true\\" class=\\"badge badge-link\\">Main Content</span>"`;

exports[`EsBadge <es-badge variant="primary" outline="false" /> 1`] = `"<span class=\\"badge badge-primary\\">Main Content</span>"`;

exports[`EsBadge <es-badge variant="primary" outline="true" /> 1`] = `"<span outline=\\"true\\" class=\\"badge badge-primary\\">Main Content</span>"`;

exports[`EsBadge <es-badge variant="secondary" outline="false" /> 1`] = `"<span class=\\"badge badge-secondary\\">Main Content</span>"`;

exports[`EsBadge <es-badge variant="secondary" outline="true" /> 1`] = `"<span outline=\\"true\\" class=\\"badge badge-secondary\\">Main Content</span>"`;

exports[`EsBadge <es-badge variant="warning" outline="false" /> 1`] = `"<span class=\\"badge badge-warning\\">Main Content</span>"`;

exports[`EsBadge <es-badge variant="warning" outline="true" /> 1`] = `"<span outline=\\"true\\" class=\\"badge badge-warning\\">Main Content</span>"`;
